hangman.py

```diff
@@ -152,62 +253,118 @@ def show_possible_matches(my_word):
+
+    * At the start of the game, let the user know how many
       letters the secret_word contains and how many guesses s/he starts with.
-      
+
     * The user should start with 6 guesses
-    
+
     * Before each round, you should display to the user how many guesses
       s/he has left and the letters that the user has not yet guessed.
-    
+
     * Ask the user to supply one guess per round. Make sure to check that the user guesses a letter
-      
-    * The user should receive feedback immediately after each guess 
+
+    * The user should receive feedback immediately after each guess
       about whether their guess appears in the computer's word.
 
-    * After each guess, you should display to the user the 
+    * After each guess, you should display to the user the
       partially guessed word so far.
-      
+
     * If the guess is the symbol *, print out all words in wordlist that
-      matches the current guessed word. 
-    
+      matches the current guessed word.
+
     Follows the other limitations detailed in the problem write-up.
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-# When you've completed your hangman_with_hint function, comment the two similar
-# lines above that were used to run the hangman function, and then uncomment
-# these two lines and run this file to test!
-# Hint: You might want to pick your own secret_word while you're testing.
+    letters_guessed = []
+    guesses = 0
+    warnings = 0
+    unique_letters = ''
+    for letter in secret_word:
+        if letter not in unique_letters:
+            unique_letters += letter
+
+    print('Welcome to the game Hangman!')
+    print('I\'m thinking of a word that is ' + str(len(secret_word)) + ' letters long.')
+    print('-----------------')
+
+    while guesses < 6:
+        letters_left = get_available_letters(letters_guessed)
+        print('You have ' + str(6 - guesses) + ' guesses left.')
+        print('Available letters: ' + letters_left)
+
+        user_guess = (input('Please guess a letter: '))
+
+        if str.isalpha(user_guess):
+            if str.isupper(user_guess):
+                user_guess = str.lower(user_guess)
+            if user_guess in letters_guessed:
+                warnings += 1
+                if warnings < 4:
+                    print('Oops! You\'ve already guessed that letter.You now have ' + str(warnings)
+                          + 'left:' + get_guessed_word(secret_word, letters_guessed))
+                elif warnings == 4:
+                    print('Oops! You\'ve already guessed that letter.You now have no warnings left:\n '
+                          'so you lose one guess:' + get_guessed_word(secret_word, letters_guessed))
+                    guesses += 1
+                else:
+                    break
+
+            else:
+                letters_guessed.append(user_guess)
+                if is_word_guessed(user_guess, secret_word):
```

Comment:
For the future reviewers of your code it will be a problem to understand, why you call the `is_word_guessed()` in order to check the inputted letter. Create an additinal function for this purpose.